build/ghactions/pr-codecov-run-tests.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -xeuo pipefail
+
+output_json_file="$1"
+packages="$2"
+
+if [ -z "${packages}" ]; then
+  echo "Skipping"
+  touch "${output_json_file}"
+  exit 0
+fi
+
+
+# Find the targets. We need to convert from, e.g.
+#   ./pkg/util/log/logpb ./pkg/util/quotapool
+# to
+#   //pkg/util/log/logpb:* + //pkg/util/quotapool:*
+
+paths=""
+sep=""
+
+for p in ${packages}; do
+  if [[ $p == ./pkg/* ]]; then
+    path="//pkg/${p#./pkg/}"
+    paths="${paths}${sep}${path}:*"
+    sep=" + "
+  else
+    echo "ignoring $p"
+  fi
+done
+
+targets=$(bazel query "kind(\".*_test\", ${paths})")
+
+if [[ -z "${targets}" ]]; then
+  echo "No test targets found"
+  exit 0
+fi
+
+echo "Running tests"
+
+bazel coverage --compilation_mode=fastbuild \
```

Comment:
`fastbuild` is the default, no reason to specify it. Note that this also results in *stripped* output. Is that undesirable for coverage? Maybe it doesn't matter, I'm not sure. If it does matter you can set `--strip=never`.